tests/end-to-end/upgrade/pkg/tests/function/function.go

```diff
@@ -0,0 +1,176 @@
+package function
+
+import (
+	"bytes"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"time"
+	"strings"
+
+	"github.com/google/uuid"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+	kubelessV1 "github.com/kubeless/kubeless/pkg/apis/kubeless/v1beta1"
+	kubeless "github.com/kubeless/kubeless/pkg/client/clientset/versioned"
+	"github.com/sirupsen/logrus"
+
+	restclient "k8s.io/client-go/rest"
+)
+
+type FunctionUpgradeTest struct {
+	functionName, uuid string
+	kubelessClient     *kubeless.Clientset
+	coreClient         *kubernetes.Clientset
+}
+
+func NewFunctionUpgradeTest(config *restclient.Config) (*FunctionUpgradeTest) {
+
+	kubelessClient, err := kubeless.NewForConfig(config)
+	if err != nil {
+		return &FunctionUpgradeTest{}
+	}
+
+	coreClient, err := kubernetes.NewForConfig(config)
+	if err != nil {
+		return &FunctionUpgradeTest{}
+	}
+
+	return &FunctionUpgradeTest{
+		kubelessClient: kubelessClient,
+		coreClient:     coreClient,
+		functionName:   "hello",
+		uuid:           uuid.New().String(),
+	}
+}
+
+func (f *FunctionUpgradeTest) CreateResources(stop <-chan struct{}, log logrus.FieldLogger, namespace string) error {
+	log.Println("FunctionUpgradeTest creating resources")
+	_, err := f.createFunction(log, namespace)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (f *FunctionUpgradeTest) TestResources(stop <-chan struct{}, log logrus.FieldLogger, namespace string) error {
+	log.Println("FunctionUpgradeTest testing resources")
+	err := f.getFunctionPodStatus(namespace, 10*time.Minute)
+	if err != nil {
+		return err
+	}
+
+	host := fmt.Sprintf("http://%s.%s:8080", namespace, namespace)
+
+	value, err := f.getFunctionOutput(host, 2*time.Minute)
```

Comment:
I couldn't pass this step on cluster with error: `Could not get function output:\n Post http://hello.functionupgradetest:8080`